typing/typedecl.ml

```diff
@@ -2192,10 +2192,29 @@ let report_error_doc ppf = function
   | Unavailable_type_constructor p ->
       fprintf ppf "The definition of type %a@ is unavailable"
         (Style.as_inline_code Printtyp.path) p
-  | Variance Typedecl_variance.Varying_anonymous ->
-      fprintf ppf "@[%s@ %s@ %s@]"
-        "In this GADT definition," "the variance of some parameter"
-        "cannot be checked"
+  | Variance (Typedecl_variance.Varying_anonymous (n, reason)) ->
+      let reason_text =
+        match reason with
+        | Variable_constrained ty ->
+            dprintf
+              "the type variable %a appears@ in other parameters.@ \
+               Co- or contra-variant@ type parameters@ \
```

Comment:
I think it might be better to keep covariant and contravariant as full words (for the sake of people not overly familiar with the terms). Also, it might be a good idea to reinforce the point that this a GADT restriction
```
In GADTs, covariant or contravariant type parameters@ 
```